stackslib/src/config/mod.rs

```diff
@@ -1656,52 +1656,220 @@ impl BurnchainConfigFile {
+    ///
+    /// Default: `"helium-node"`
     pub name: String,
+    /// The node's Bitcoin wallet private key, provided as a hex string in the config file.
+    /// Used to initialize the node's keychain for signing operations.
+    /// If `miner.mining_key` is not set, this seed may also be used for mining-related signing.
+    /// Required if the node is configured as a miner (`node.miner = true`) and `miner.mining_key` is absent.
+    ///
+    /// Default: Randomly generated 32 bytes.
     pub seed: Vec<u8>,
+    /// The file system absolute path to the node's working directory.
+    /// All persistent data, including chainstate, burnchain databases, and potentially other stores,
+    /// will be located within this directory.
+    /// This path can be overridden by setting the `STACKS_WORKING_DIR` environment variable.
+    ///
+    /// Default: `/tmp/stacks-node-{current_timestamp}`.
     pub working_dir: String,
+    /// The IPv4 address and port (e.g., "0.0.0.0:20443") on which the node's HTTP RPC server
+    /// should bind and listen for incoming API requests.
+    ///
+    /// Default: `"0.0.0.0:20443"`
     pub rpc_bind: String,
+    /// The IPv4 address and port (e.g., "0.0.0.0:20444") on which the node's P2P networking
+    /// service should bind and listen for incoming connections from other peers.
+    ///
+    /// Default: `"0.0.0.0:20444"`
     pub p2p_bind: String,
+    /// The publicly accessible URL that this node advertises to peers during the P2P handshake
+    /// as its HTTP RPC endpoint. Other nodes or services might use this URL to query the node's API.
+    ///
+    /// Default: `http://{rpc_bind}` (e.g., "http://127.0.0.1:20443" if rpc_bind is default and resolves locally).
     pub data_url: String,
+    /// The publicly accessible IPv4 address and port that this node advertises to peers for P2P connections.
+    /// This might differ from `p2p_bind` if the node is behind NAT or a proxy.
+    /// Note: The default value derivation might be unexpected, potentially using the `rpc_bind` address; explicit configuration is recommended if needed.
+    ///
+    /// Default: Derived from `rpc_bind` (e.g., "127.0.0.1:20443" with default settings).
     pub p2p_address: String,
+    /// The private key seed, provided as a hex string in the config file, used specifically for the
+    /// node's identity and message signing within the P2P networking layer.
+    /// This is separate from the main `node.seed`.
+    ///
+    /// Default: Randomly generated 32 bytes.
     pub local_peer_seed: Vec<u8>,
+    /// A list of initial peer nodes used to bootstrap connections into the Stacks P2P network.
+    /// Peers are specified as comma-separated strings in the format "PUBKEY@IP:PORT" or "PUBKEY@HOSTNAME:PORT"
+    /// in the configuration file. DNS hostnames are resolved during configuration loading.
+    /// If empty on mainnet, default Hiro and Stacks seed nodes are used.
+    ///
+    /// Default: Empty vector `[]`.
```

Comment:
There are definitely default instances for this setting - i'll have to look it up, but if you do not define this key in the toml, the defaults are used. it might not be worth listing what the defaults are - it's a long string

Reply:
https://github.com/stacks-network/stacks-core/blob/develop/stackslib/src/config/mod.rs#L265